chromium_src/third_party/blink/renderer/modules/broadcastchannel/broadcast_channel.cc

```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/public/mojom/broadcastchannel/broadcast_channel.mojom-blink.h"
+#include "third_party/blink/renderer/modules/storage/brave_dom_window_storage.h"
+
+#define ConnectToChannel                                                   \
+  Version_;                                                                \
+  {                                                                        \
+    LocalDOMWindow* window = DynamicTo<LocalDOMWindow>(execution_context); \
+    if (window) {                                                          \
```

Comment:
what if this is not a window, but a worker?

Reply:
Worker will use a valid origin. Any ephemeral RFH still wouldn't be able to communicate with this worker (in both directions) because a frame-based ```BroadcastChannel``` will use an ephemeral origin instead of the one that worker is using.
In 1PES context we might want to disable all workers.

Reply:
gotcha, please add a comment

Reply:
but wouldn't it still be useful to have `GetEphemeralStorageOrigin` take ExecutionContext instead of `LocalDOMWindow` so it can work for both?

Reply:
> but wouldn't it still be useful to have GetEphemeralStorageOrigin take ExecutionContext instead of LocalDOMWindow so it can work for both?

I actually wanted to do that but it still be usable only for DOMWindow-specific ExecutionContexts. In other words, it will pretend that it supports any ExecutionContext but in reality it's not.
Worker-related ExecutionContexts use different ```WebContentSettingsClient``` implementation.

Let's keep it this way for now to make our implementation clear.

Reply:
@goodov `GetContentSettingsClientFor` ? That's what we use in Farbling for workers cc @pilgrim-brave

Reply:
> @goodov ```GetContentSettingsClientFor```? That's what we use in Farbling for workers cc @pilgrim-brave

We need to implement ```GetEphemeralStorageOrigin``` in ```WorkerContentSettingsClient``` to be able to use ```GetContentSettingsClientFor(...)->GetEphemeralStorageOrigin``` everywhere. Currently it will return an empty origin in Worker-related ExecutionContexts, that's why I decided to not use ```GetContentSettingsClientFor``` here.

I think we can go that way and add support for ephemeral origins for Worker-related ExecutionContexts, but it looks like a separate task.